packages/graphql/src/services/notifications.ts

```diff
@@ -49,3 +51,130 @@ export async function subscribeToPush(
     },
   })
 }
+
+interface UnsubscribeToPushArgs {
+  courseId: string
+  endpoint: string
+}
+
+export async function unsubscribeFromPush( 
+  { courseId, endpoint }: UnsubscribeToPushArgs,
+  ctx: ContextWithUser
+) {
+  try {
+    await ctx.prisma.pushSubscription.delete({
+      where: {
+        participantId_courseId_endpoint: {
+          participantId: ctx.user.sub,
+          courseId,
+          endpoint
+        }
+      }
+    })
+    return true
+  } catch(error) {
+    console.error("An error occured while trying to unsubscribe from push notifications: ", error)
+    return false
+  }
+}
+
+export async function sendPushNotifications( ctx: Context) {
+  webpush.setVapidDetails(
+    'mailto:klicker.support@uzh.ch',
+    process.env.VAPID_PUBLIC_KEY as string,
+    process.env.VAPID_PRIVATE_KEY as string
+  )
+
+  const microSessions = await ctx.prisma.microSession.findMany({
+    where: {
+      status: MicroSessionStatus.PUBLISHED,
+      scheduledStartAt: {
+        lte: new Date(),
+      },
+      scheduledEndAt: {
+        gt: new Date(),
+      },
+      arePushNotificationsSent: false,
+    },
+    include: {
+      course: {
+        include: {
+          subscriptions: true,
+        },
+      },
+      instances: false,
+    },
+  })
+
+
+  await Promise.all(
+    microSessions.map(async (microSession: MicroSession) => {
```

Comment:
we will need to think about scaling of this. could it lead to issues down the line when many people create a lot of micro sessions? we should balance the execution count of the cronjob (e.g., every 5 minutes) against the number of created micro sessions per hour or similar

Reply:
it could potentially be extracted into a background process